### DISCUSSION TITLE
Add function to export meshes/grids to ZMAP Grids

### DISCUSSION THREAD

**Discussion by AlexanderJuestel** [pyvista/pyvista#2255] [Ideas]
Dear all, 

we are using PyVista within the GemPy framework to create models of the subsurface. The created meshes represent the boundaries/depths between different geological layers. While the 3D representation of the meshes is great, it is more feasible to work with them in 2D in packages like QGIS/ArcGIS, especially when creating reports. 

I would like to suggest now adding a function that allows the export of the meshes as ZMAP Grids. Here is a little explanation of how the files look like: https://saurabhkukade.com/posts/2020/07/understanding-zmap-file-format/.

I would be happy to contribute in implementing the feature as we may need it quite often in the future and currently use a quite laborious alternative way.

Cheers Alex

**Comment by RichardScottOZ** [comment]
So you want an automatic depth slicer into that format, which is presumably uniform and gdal readable?

**Reply by AlexanderJuestel** [reply]
One can put it that way, yes. While we do like the 3D view, it is sometimes easier to explain data/information in a 2D view, especially for non-geoscientists. So maybe we can get something going :)

**Comment by AlexanderJuestel** [comment]
@banesullivan I have implemented a way to extract the data from a mesh using your implemented ray tracing and saving these results to a ZMAP file. Have a look at the little notebook that I created. Some feedback would be appreciated if you think that there is a better way to do it. This functionality will be implemented for the next release of GemGIS :) 

https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/54_converting_pyvista_mesh_to_zmap_grid.html

**Reply by banesullivan** [reply]
Sorry for the dalay! Generally this looks fantastic! I don't have much time/bandwidth to dive fully into it, so no feedback from me at this time. Nice work!
